feature/google/src/latestDependencies/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -155,26 +154,47 @@ class BillingWrapper(
         executeRequestOnUIThread { connectionError ->
             if (connectionError == null) {
-                val params = SkuDetailsParams.newBuilder()
-                    .setType(productType.toGoogleProductType() ?: SkuType.INAPP)
-                    .setSkusList(nonEmptySkus).build()
+                val productList = skus.map { sku ->
+                    QueryProductDetailsParams.Product.newBuilder()
+                        .setProductId(sku)
+                        .setProductType(productType.toGoogleProductType() ?: BillingClient.ProductType.SUBS)
+                        .build()
+                }
+                val params = QueryProductDetailsParams.newBuilder()
+                    .setProductList(productList).build()
 
                 withConnectedClient {
-                    querySkuDetailsAsyncEnsuringOneResponse(params) { billingResult, skuDetailsList ->
+                    queryProductDetailsAsyncEnsuringOneResponse(params) { billingResult, productDetailsList ->
                         if (billingResult.responseCode == BillingClient.BillingResponseCode.OK) {
                             log(
                                 LogIntent.DEBUG, OfferingStrings.FETCHING_PRODUCTS_FINISHED
                                     .format(skus.joinToString())
                             )
                             log(
                                 LogIntent.PURCHASE, OfferingStrings.RETRIEVED_PRODUCTS
-                                    .format(skuDetailsList?.joinToString { it.toString() })
+                                    .format(productDetailsList.joinToString { it.toString() })
                             )
-                            skuDetailsList?.takeUnless { it.isEmpty() }?.forEach {
-                                log(LogIntent.PURCHASE, OfferingStrings.LIST_PRODUCTS.format(it.sku, it))
+                            productDetailsList.takeUnless { it.isEmpty() }?.forEach {
+                                log(LogIntent.PURCHASE, OfferingStrings.LIST_PRODUCTS.format(it.productId, it))
+                            }
+
+                            val storeProducts = mutableListOf<StoreProduct>()
+                            productDetailsList.forEach { product ->
+                                val basePlans = product.subscriptionOfferDetails?.filter {
+                                    it.pricingPhases.pricingPhaseList.size == 1
+                                } ?: emptyList()
+
+                                val groupedOffers = product.subscriptionOfferDetails?.groupBy {
+                                    it.pricingPhases.pricingPhaseList.last().billingPeriod
+                                } ?: emptyMap()
+                                basePlans.takeUnless { it.isEmpty() }?.forEach { basePlan ->
+                                    val billingPeriod = basePlan.pricingPhases.pricingPhaseList[0].billingPeriod
```

Comment:
pricingPhases.pricingPhaseList.get(0)?.billingPeriod 

guarding against a potential empty list, though i don't see why that should happen